gcloud-java-compute/src/main/java/com/google/gcloud/compute/DeprecationStatus.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * The deprecation status associated to a Google Compute Engine resource.
+ */
+public final class DeprecationStatus<T extends ResourceId> implements Serializable {
+
+  private static final long serialVersionUID = -2695077634793679794L;
+
+  private final String deleted;
```

Comment:
> BTW, I would argue that consistency with the rest of our API is more important in that case.

Agree! 

> Sorry for the intermediate review. Will let you know when I am done (hopefully by EOD PST).

It's fine with intermediate reviews, I just wanted to be sure to avoid pushing intermediate commits